solr/core/src/java/org/apache/solr/update/processor/DistributedUpdateProcessor.java

```diff
@@ -1184,7 +1196,16 @@ protected boolean versionAdd(AddUpdateCommand cmd) throws IOException {
 
         // TODO: possibly set checkDeleteByQueries as a flag on the command?
         doLocalAdd(cmd);
-        
+
+        if(lastKnownVersion != null && req.getSchema().isUsableForChildDocs() &&
```

Comment:
This may very well be right but can you tell me why you added this delete here at this line and why the delete is version-dependent?

Reply:
This delete ensures the updated block is the latest block, deleting all previous ones.